### PR TITLE
[TM-861] add polygon version

### PR DESCRIPTION
**Pull request by pachonjcl**
[Task Link](https://gfw.atlassian.net/browse/TM-861)

### CODE REVIEW

File: database/migrations/2024_07_05_140225_alter_site_polygon_table_to_add_versioning_attributes.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+return new class extends Migration {
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('site_polygon', function (Blueprint $table) {
+            $table->string('version_name')->nullable();
+            $table->string('primary_uuid')->nullable()->after('uuid');
+            $table->boolean('is_active')->default(false);
```
Comment (pachonjcl) [inline database/migrations/2024_07_05_140225_alter_site_polygon_table_to_add_versioning_attributes.php:16]: This one is not a boolean and references has been updated to true / false accordingly.

File: database/migrations/2024_07_05_140225_alter_site_polygon_table_to_add_versioning_attributes.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        });
+
+        DB::statement('UPDATE site_polygon set primary_uuid = uuid, is_active = true;');
+
+        Schema::table('site_polygon', function (Blueprint $table) {
+            $table->uuid('primary_uuid')->nullable(false)->change();
+            $table->index('primary_uuid');
```
Comment (pachonjcl) [inline database/migrations/2024_07_05_140225_alter_site_polygon_table_to_add_versioning_attributes.php:23]: There is not an index on primary_uuid

File: config/database.php [hunk 3]
```diff
@@ -56,7 +56,7 @@
             'collation' => 'utf8mb4_unicode_ci',
             'prefix' => '',
             'prefix_indexes' => true,
-            'strict' => true,
+            'strict' => false,
```
Comment (pachonjcl) [inline config/database.php:59]: Making reference of this [SO post](https://stackoverflow.com/questions/47974348/sqlstatehy000-general-error-1364-field-name-doesnt-have-a-default-value-l)
Unit tests were throwing errors as follows:
`PDOException: SQLSTATE[HY000]: General error: 1364 Field 'primary_uuid' doesn't have a default value`
This is the difference between strict usage and not.

File: app/Http/Controllers/V2/Sites/UpdateSitePolygonActiveController.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public function __invoke(string $uuid): JsonResponse
+    {
+        Log::info('Fetching site polygons', ['uuid' => $uuid]);
+
+        try {
+            $sitePolygon = SitePolygon::isUuid($uuid)->first();
+            if (! $sitePolygon) {
+                return response()->json(['error' => 'Site polygon not found'], 404);
+            }
+            SitePolygon::where('primary_uuid', $sitePolygon->primary_uuid)->update(['is_active' => 0]);
```
Comment (roguenet) [inline app/Http/Controllers/V2/Sites/UpdateSitePolygonActiveController.php:21]: This should use `=> false`

File: app/Models/V2/Sites/SitePolygon.php [hunk 5]
```diff
@@ -97,4 +101,39 @@ public function getAuditableNameAttribute(): string
     {
         return $this->poly_name ?? '';
     }
+
+    public function scopeActive(Builder $query)
+    {
+        return $query->where('is_active', 1);
```
Comment (roguenet) [inline app/Models/V2/Sites/SitePolygon.php:107]: `where('is_active', true)`

File: database/migrations/2024_07_05_140225_alter_site_polygon_table_to_add_versioning_attributes.php (hunk 1 above)
Comment (roguenet) [inline database/migrations/2024_07_05_140225_alter_site_polygon_table_to_add_versioning_attributes.php:16]: On reading this again, I think maybe the default should be true?

File: app/Models/V2/Sites/SitePolygon.php (hunk 5 above)
Comment (pachonjcl) [reply app/Models/V2/Sites/SitePolygon.php]: changed

File: app/Http/Controllers/V2/Sites/UpdateSitePolygonActiveController.php (hunk 4 above)
Comment (pachonjcl) [reply app/Http/Controllers/V2/Sites/UpdateSitePolygonActiveController.php]: changed

**Review by roguenet — APPROVED** [review]

**Pull request merged**